app/ante_gate.go

```diff
@@ -0,0 +1,43 @@
+package app
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+type AnteSpamMigitationDecorator struct {
+	pk PylonsKeeper
+}
+
+func NewSpamMigitationAnteDecorator(pylonsmodulekeeper PylonsKeeper) AnteSpamMigitationDecorator {
+	return AnteSpamMigitationDecorator{
+		pk: pylonsmodulekeeper,
+	}
+}
+
+// AnteDecorator
+func (ad AnteSpamMigitationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if (ctx.IsCheckTx() || ctx.IsReCheckTx()) && !simulate {
+		sigTx, ok := tx.(authsigning.SigVerifiableTx)
+		if !ok {
+			return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "invalid transaction type")
+		}
+
+		// get max txs in a block, default is 5
```

Comment:
@faddat as @mustafapylons and I reviewed the code we find following things

1. By default it use 20 transaction as max transaction, is there any reason for choosing this number, also can we update this comment as it says 5 by default at line 29

2. What if there are 2 users 1 send 20 transactions in block and other use send only 4, so are we going to reject the whole block or only the user 1 who is spamming, and can we write some test cases to ensure this situation. 

3. can we remove dead code on line 30, // ante.NewMempoolFeeDecorator()

Reply:
Thanks for reviewing. This is my answer to your question:
 1. Number of transactions in the block is a param. You can change it in genesis or by a gov. 
 2. Max transactions in a block are a metric per user. It means if user 1 sent 20 transactions in a block (max transactions in a block), user2 could be continue sent
 3. Yes, make code cleaner!